src/main/java/org/prebid/server/bidder/huaweiads/HuaweiUtilRequest.java

```diff
@@ -0,0 +1,474 @@
+        }
+
+        try {
+            extUserDataHuaweiAds = mapper.mapper().readValue(userExt.getData().toString(),
+                    ExtUserDataHuaweiAds.class);
+        } catch (JsonProcessingException e) {
+            throw new PreBidException("Unmarshal: BidRequest.user.ext -> extUserDataHuaweiAds failed");
+        }
+        return extUserDataHuaweiAds.getData();
+    }
+
+    private static String resolveCountry(BidRequest bidRequest) {
+        Geo geo;
+        String country;
+
+        Device device = bidRequest.getDevice();
+        geo = device != null ? device.getGeo() : null;
+        country = geo != null ? geo.getCountry() : null;
+        if (!StringUtils.isBlank(country)) {
+            return convertCountryCode(country);
+        }
+
+        User user = bidRequest.getUser();
+        geo = user != null ? user.getGeo() : null;
+        country = geo != null ? geo.getCountry() : null;
+        if (!StringUtils.isBlank(country)) {
+            return convertCountryCode(country);
+        }
+
+        return DEFAULT_COUNTRY_NAME;
+    }
+
+    private static String convertCountryCode(String country) {
+        Map<String, String> mapCountryCodeAlpha3ToAlpha2 = new HashMap<>();
+        mapCountryCodeAlpha3ToAlpha2.put("CHL", "CL");
+        mapCountryCodeAlpha3ToAlpha2.put("CHN", "CN");
+        mapCountryCodeAlpha3ToAlpha2.put("ARE", "AE");
+        String countryPresentInMap = mapCountryCodeAlpha3ToAlpha2.keySet().stream()
+                .filter(element -> element.equals(country))
+                .findFirst()
+                .orElse(null);
+
+        if (countryPresentInMap != null) {
+            return mapCountryCodeAlpha3ToAlpha2.get(countryPresentInMap);
+        }
+        if (country.length() > 2) {
+            return country.substring(0, 2);
+        }
+        return DEFAULT_COUNTRY_NAME;
+    }
+
+    private static String resolvePkgName(App appFromBidReq) {
+        String bundle = appFromBidReq.getBundle();
+        if (StringUtils.isBlank(bundle)) {
+            throw new PreBidException("resolvePkgName: app.bundle is empty");
+        }
+        return bundle;
+    }
+
+    private static String resolveLang(App appFromBidReq) {
+        String lang = appFromBidReq.getContent() != null ? appFromBidReq.getContent().getLanguage() : null;
+        if (StringUtils.isBlank(lang)) {
+            return "en";
+        }
+        return lang;
+    }
+
+    private static HuaweiNetwork resolveHuaweiNetwork(BidRequest bidRequest) {
+        Device device = bidRequest.getDevice();
+        if (device != null) {
+            return HuaweiNetwork.builder()
+                    .type(device.getConnectiontype() != null
+                            ? device.getConnectiontype()
+                            : DEFAULT_UNKNOWN_NETWORK_TYPE)
+                    .carrier(resolveCarrier(device))
+                    .cellInfoList(resolveCellInfoList(device))
+                    .build();
+        }
+        return null;
```

Comment:
invert condition to move this bunch of code out of if statement and make code clearer